src/AddressSpace.h

```diff
@@ -829,6 +840,15 @@ class AddressSpace : public HasTaskSet {
 
   bool legacy_breakpoint_mode() { return stopping_breakpoint_table_ != nullptr; }
   remote_code_ptr do_breakpoint_fault_addr() { return do_breakpoint_fault_addr_; }
+
+  void set_breakpoint_fault_addr(remote_code_ptr addr) {
+    do_breakpoint_fault_addr_ = addr;
+  }
+
+  void set_uses_syscall_buffer(bool uses_syscall_buffer = true) {
+    syscallbuf_enabled_ = uses_syscall_buffer;
+  }
```

Comment:
I think we'd benefit from a method on AddressSpace that restores auxv, do_breakpoint_fault_addr_, and syscallbuf_enabled_ all together.